tests/lib/cli.js

```diff
@@ -1006,6 +1006,18 @@ describe("cli", () => {
                 assert.isTrue(log.info.notCalled);
             });
 
+            // https://github.com/eslint/eslint/issues/17660
+            it(`should write the file and create dirs if they don't exist even when output is empty with configType:${configType}`, async () => {
+                const filePath = getFixturePath("single-quoted.js");
+                const code = `${flag} --rule 'quotes: [1, single]' --o tests/output/eslint-output.txt ${filePath}`;
+
+                await cli.execute(code, "var a = 'b'", useFlatConfig);
```

Comment:
```suggestion
                // TODO: fix this test to: await cli.execute(code, null, useFlatConfig); 
                await cli.execute(code, "var a = 'b'", useFlatConfig);
```

We should fix all tests that end up targeting files outside the cwd (in particular, we should change the cwd), because in flat config mode those files are ignored. For now, I think it's fine to use stdin here and leave a TODO comment.